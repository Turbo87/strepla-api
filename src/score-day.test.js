const fs = require('fs');
const { parseScoreDay } = require('./score-day');

describe('scoreDay', () => {
  const TESTS = [
    'stoelln17-std-day1',
  ];

  for (let fixtureName of TESTS) {
    test(fixtureName, () => {
      let body = fs.readFileSync(`${__dirname}/../__fixtures__/${fixtureName}.html`);
      expect(parseScoreDay(body)).toMatchSnapshot();
    });
  }
});
