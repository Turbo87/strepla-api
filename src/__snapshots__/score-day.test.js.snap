// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`scoreDay stoelln17-std-day1 1`] = `
Object {
  "class": "Std",
  "competition": Object {
    "description": "SLP Stölln/Rhinow 7/27/2017-8/3/2017",
    "name": "Gollenberg Cup 2017",
  },
  "date": "2017-07-28",
  "remark": "Pmax=601,3 Pdm=267,2 Pvm=334,0 Do=198,9 f=1,000 N=18 n1=16 n2=15
(*)=scoring time equal Tmin
dnf=did not fly",
  "results": Array [
    Object {
      "competitionId": "EX",
      "country": "BB",
      "distance": 198.9,
      "glider": "LS8",
      "name": "Gäde, Julius",
      "penaltyPoints": 0,
      "points": 601,
      "rank": 1,
      "speed": 27.577777777777776,
      "taskStartTime": "13:07:15",
      "time": 7211,
    },
    Object {
      "competitionId": "8A",
      "country": "BB",
      "distance": 194.4,
      "glider": "LS8neo",
      "name": "Tiebel, Daniel",
      "penaltyPoints": 0,
      "points": 580,
      "rank": 2,
      "speed": 26.99722222222222,
      "taskStartTime": "13:07:25",
      "time": 7117,
    },
    Object {
      "competitionId": "LB",
      "country": "BL",
      "distance": 184.7,
      "glider": "LS8",
      "name": "Blattmann, Lukas",
      "penaltyPoints": 0,
      "points": 517,
      "rank": 3,
      "speed": 25.25,
      "taskStartTime": "13:06:16",
      "time": 7315,
    },
    Object {
      "competitionId": "GP",
      "country": "BY",
      "distance": 179.1,
      "glider": "LS8neo",
      "name": "Lauer, Philipp",
      "penaltyPoints": 0,
      "points": 491,
      "rank": 4,
      "speed": 24.544444444444444,
      "taskStartTime": "13:08:03",
      "time": 7296,
    },
    Object {
      "competitionId": "73",
      "country": "NRW",
      "distance": 181,
      "glider": "Discus 2b",
      "name": "Kwiatkowski, Ron",
      "penaltyPoints": 0,
      "points": 474,
      "rank": 5,
      "speed": 24.080555555555556,
      "taskStartTime": "13:08:23",
      "time": 7518,
    },
    Object {
      "competitionId": "NW1",
      "country": "NRW",
      "distance": 173.4,
      "glider": "Discus 2b",
      "name": "Langner, Thorben",
      "penaltyPoints": 0,
      "points": 462,
      "rank": 6,
      "speed": 23.75,
      "taskStartTime": "13:07:08",
      "time": 7301,
    },
    Object {
      "competitionId": "BW3",
      "country": "BW",
      "distance": 189.7,
      "glider": "Discus 2b",
      "name": "Hiller, Chris",
      "penaltyPoints": 0,
      "points": 456,
      "rank": 7,
      "speed": 23.580555555555556,
      "taskStartTime": "13:11:36",
      "time": 8047,
    },
    Object {
      "competitionId": "X",
      "country": "NI",
      "distance": 173.7,
      "glider": "LS8",
      "name": "Ahrens, Wilm",
      "penaltyPoints": 0,
      "points": 436,
      "rank": 8,
      "speed": 23.033333333333335,
      "taskStartTime": "13:07:40",
      "time": 7540,
    },
    Object {
      "competitionId": "GO",
      "country": "BW",
      "distance": 161.8,
      "glider": "Discus 2b",
      "name": "Linkner, Lars",
      "penaltyPoints": 0,
      "points": 416,
      "rank": 9,
      "speed": 22.469444444444445,
      "taskStartTime": "13:07:59",
      "time": 6879,
    },
    Object {
      "competitionId": "WS",
      "country": "NRW",
      "distance": 157.8,
      "glider": "LS8",
      "name": "Löser, Hendrik",
      "penaltyPoints": 0,
      "points": 395,
      "rank": 10,
      "speed": 21.913888888888888,
      "taskStartTime": "13:07:56",
      "time": 6789,
    },
    Object {
      "competitionId": "TX",
      "country": "NI",
      "distance": 161.7,
      "glider": "LS8",
      "name": "Polej, Dennis",
      "penaltyPoints": 0,
      "points": 341,
      "rank": 11,
      "speed": 20.42777777777778,
      "taskStartTime": "13:12:57",
      "time": 7914,
    },
    Object {
      "competitionId": "S1",
      "country": "RP",
      "distance": 163,
      "glider": "LS8",
      "name": "Dibbern, Johannes",
      "penaltyPoints": 0,
      "points": 341,
      "rank": 11,
      "speed": 20.419444444444444,
      "taskStartTime": "13:12:53",
      "time": 7983,
    },
    Object {
      "competitionId": "BI",
      "country": "HE",
      "distance": 160.8,
      "glider": "LS8",
      "name": "Schlautmann, Nils",
      "penaltyPoints": 0,
      "points": 328,
      "rank": 13,
      "speed": 20.05833333333333,
      "taskStartTime": "13:10:44",
      "time": 8019,
    },
    Object {
      "competitionId": "PL",
      "country": "RP",
      "distance": 161.9,
      "glider": "ASW28",
      "name": "Frey, Yannick",
      "penaltyPoints": 0,
      "points": 327,
      "rank": 14,
      "speed": 20.033333333333335,
      "taskStartTime": "13:12:51",
      "time": 8082,
    },
    Object {
      "competitionId": "70",
      "country": "NRW",
      "distance": 160.3,
      "glider": "LS8neo",
      "name": "Fecker, Nils",
      "penaltyPoints": 0,
      "points": 324,
      "rank": 15,
      "speed": 19.95,
      "taskStartTime": "13:10:27",
      "time": 8037,
    },
    Object {
      "competitionId": "4J",
      "country": "BY",
      "distance": 135.8,
      "glider": "Discus 2c",
      "name": "Heinig, Jan",
      "penaltyPoints": 0,
      "points": 267,
      "rank": 16,
      "speed": 17.055555555555554,
      "taskStartTime": "13:18:02",
      "time": 7964,
    },
    Object {
      "competitionId": "HW",
      "country": "NRW",
      "distance": 48.6,
      "glider": "LS8neo",
      "name": "Klomp, Wilhelm-Alexander",
      "penaltyPoints": 0,
      "points": 65,
      "rank": 17,
      "speed": null,
      "taskStartTime": "13:07:50",
      "time": null,
    },
    Object {
      "competitionId": "TM",
      "country": "NI",
      "distance": 27.8,
      "glider": "Discus 2c",
      "name": "Josuttis, Benjamin",
      "penaltyPoints": 0,
      "points": 37,
      "rank": 18,
      "speed": null,
      "taskStartTime": "13:13:28",
      "time": null,
    },
  ],
  "taskDescription": "Tagesaufgabe A - Speed Assigned Area Task (AAT) (Tmin=02:00)  82.0 km < D < 429.8 km:  000Abflugpunkt Nord-> 068Lenzen B1-> 053Graeben-> 003Anflugpunkt Ost-> 121Stoelln",
}
`;
